### PR TITLE
Staging to master

### PR DESCRIPTION
**Pull request by Phaedra144**

### CODE REVIEW

File: build/reports/tests/test/css/base-style.css [hunk 1]
```diff
@@ -0,0 +1,179 @@
+
+body {
```
Comment (adamgyulavari) [inline build/reports/tests/test/css/base-style.css]: `build` folder should be gitignored

File: src/main/java/com/greenfoxacademy/opal/kalendaryo/kalendaryo/authorization/AuthorizeKal.java [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+
+    public static Calendar getCalendarService() throws IOException {
+        Credential credential = authorize();
+        return new com.google.api.services.calendar.Calendar.Builder(
+                HTTP_TRANSPORT, JSON_FACTORY, credential)
+                .setApplicationName(APPLICATION_NAME)
+                .build();
+    }
+}
+
```
Comment (adamgyulavari) [inline src/main/java/com/greenfoxacademy/opal/kalendaryo/kalendaryo/authorization/AuthorizeKal.java]: we dont need this right now on staging or on production

File: src/main/java/com/greenfoxacademy/opal/kalendaryo/kalendaryo/controllers/CalendarController.java [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+    }
+
+    @DeleteMapping("/delete")
+    public String deleteCalendar() throws IOException {
+        com.google.api.services.calendar.Calendar service =
+                getCalendarService();
+        service.calendars().delete("secondaryCalendarId").execute();
+        return "redirect:https://calendar.google.com/calendar/b/1/r";
+    }
+}
```
Comment (adamgyulavari) [inline src/main/java/com/greenfoxacademy/opal/kalendaryo/kalendaryo/controllers/CalendarController.java]: we dont need this as well

File: src/main/java/com/greenfoxacademy/opal/kalendaryo/kalendaryo/authorization/AuthorizeKal.java [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+                        .setDataStoreFactory(DATA_STORE_FACTORY)
+                        .setAccessType("offline")
+                        .build();
+        Credential credential = new AuthorizationCodeInstalledApp(
+                flow, new LocalServerReceiver()).authorize("user");
+        return credential;
+    }
+
+    public static Calendar getCalendarService() throws IOException {
+        Credential credential = authorize();
```
Comment (adamgyulavari) [inline src/main/java/com/greenfoxacademy/opal/kalendaryo/kalendaryo/authorization/AuthorizeKal.java]: here you should create the `Credential` object with an `access_token` parameter

File: src/main/java/com/greenfoxacademy/opal/kalendaryo/kalendaryo/controllers/EventController.java [hunk 5]
```diff
@@ -0,0 +1,100 @@
...
+        String pageToken = null;
+        do {
+            Events events = service.events().list("primary").setPageToken(pageToken).execute();
+            List<Event> items = events.getItems();
+            for (Event event : items) {
+                System.out.println(event.getSummary());
+            }
+            pageToken = events.getNextPageToken();
+        } while (pageToken != null);
+        return "redirect:https://calendar.google.com/calendar/b/1/r";
```
Comment (adamgyulavari) [inline src/main/java/com/greenfoxacademy/opal/kalendaryo/kalendaryo/controllers/EventController.java]: something like this should work with the `access_token` query parameter and it should return the events json object

File: src/main/java/com/greenfoxacademy/opal/kalendaryo/kalendaryo/controllers/NotificationController.java [hunk 6]
```diff
@@ -0,0 +1,34 @@
...
+
+        return eventResponse;
+    }
+
+    @GetMapping(value = "/allnotis")
+    public Iterable<EventResponse> showAllEventResponse() {
+        Iterable<EventResponse> responses = eventResponseService.findAllEventResponse();
+        System.out.println(responses);
+        return responses;
+    }
```
Comment (adamgyulavari) [inline src/main/java/com/greenfoxacademy/opal/kalendaryo/kalendaryo/controllers/NotificationController.java]: cool! 👍

File: src/main/java/com/greenfoxacademy/opal/kalendaryo/kalendaryo/controllers/NotificationController.java [hunk 7]
```diff
@@ -0,0 +1,34 @@
...
+    EventResponseService eventResponseService;
+
+    @PostMapping(value = "/noti")
+    public EventResponse eventNotification(@RequestBody EventResponse eventResponse) {
+
+        System.out.println("The kind of the response: " + eventResponse.getKind() + "\n The ID of the notification channel: " + eventResponse.getId() + "\n The ID of the watched event" + eventResponse.getResourceId() + "\n The resourceUri of the watch: " + eventResponse.getResourceUri());
+
+        eventResponseService.saveEventResponse(eventResponse);
+
+        return eventResponse;
```
Comment (adamgyulavari) [inline src/main/java/com/greenfoxacademy/opal/kalendaryo/kalendaryo/controllers/NotificationController.java]: I think you dont need to send back the received event, you just respond with plain 200 OK

Comment (adamgyulavari) [reply src/main/java/com/greenfoxacademy/opal/kalendaryo/kalendaryo/controllers/NotificationController.java]: `new HttpResponse(HttpStatus.OK);`

File: src/main/java/com/greenfoxacademy/opal/kalendaryo/kalendaryo/service/EventModelService.java [hunk 8]
```diff
@@ -0,0 +1,39 @@
...
+
+        return eventModel;
+    }
+
+    public void saveEvent(EventModel eventModel) {
+        eventModelRepository.save(eventModel);
+    }
+
+
+}
```
Comment (adamgyulavari) [inline src/main/java/com/greenfoxacademy/opal/kalendaryo/kalendaryo/service/EventModelService.java]: remove unnecessary models, repos and services

File: src/main/java/com/greenfoxacademy/opal/kalendaryo/kalendaryo/KalendaryoApplication.java [hunk 9]
```diff
@@ -8,5 +8,7 @@
 
 	public static void main(String[] args) {
 		SpringApplication.run(KalendaryoApplication.class, args);
+
```
Comment (adamgyulavari) [inline src/main/java/com/greenfoxacademy/opal/kalendaryo/kalendaryo/KalendaryoApplication.java]: unnecessary empty line

File: src/main/java/com/greenfoxacademy/opal/kalendaryo/kalendaryo/KalendaryoApplication.java [hunk 10]
```diff
@@ -8,5 +8,7 @@
...
 	}
+
```
Comment (adamgyulavari) [inline src/main/java/com/greenfoxacademy/opal/kalendaryo/kalendaryo/KalendaryoApplication.java]: again

File: src/main/java/com/greenfoxacademy/opal/kalendaryo/kalendaryo/controllers/NotificationController.java [hunk 11]
```diff
@@ -0,0 +1,59 @@
...
+        if (eventResponse.validate().equals(HttpStatus.OK)) {
+            eventResponseService.saveEventResponse(eventResponse);
+        } else {
+            System.out.println("Missing: " + eventResponse.getMissingFields());
+        }
+
+        return eventResponse.validate();
+    }
+
+    @GetMapping(value = "/allnotifications")
```
Comment (adamgyulavari) [inline src/main/java/com/greenfoxacademy/opal/kalendaryo/kalendaryo/controllers/NotificationController.java]: let's just call this GET `/notification`

File: src/main/java/com/greenfoxacademy/opal/kalendaryo/kalendaryo/model/EventResponse.java [hunk 12]
```diff
@@ -0,0 +1,146 @@
...
+import javax.persistence.GenerationType;
+import javax.persistence.Id;
+import javax.servlet.http.HttpServletRequest;
+
+@Entity
+public class EventResponse {
+
+    @Id
+    @GeneratedValue(strategy = GenerationType.AUTO)
+    Long eventResponseId;
```
Comment (adamgyulavari) [inline src/main/java/com/greenfoxacademy/opal/kalendaryo/kalendaryo/model/EventResponse.java]: nope, if its autogenerated, lets just call it `id`

File: src/main/java/com/greenfoxacademy/opal/kalendaryo/kalendaryo/model/EventResponse.java [hunk 13]
```diff
@@ -0,0 +1,146 @@
...
+
+    public String getChannelToken() {
+        return channelToken;
+    }
+
+    public void setChannelToken(String channelToken) {
+        this.channelToken = channelToken;
+    }
+
+    public HttpStatus validate() {
```
Comment (adamgyulavari) [inline src/main/java/com/greenfoxacademy/opal/kalendaryo/kalendaryo/model/EventResponse.java]: validate should decide if the model data is valid, the return type should be boolean, then in the controller you can either do this or that

File: build.gradle [hunk 14]
```diff
@@ -15,21 +15,23 @@ apply plugin: 'eclipse'
...
 
-
 dependencies {
 	compile('org.springframework.boot:spring-boot-starter-data-jpa')
 	compile('org.springframework.boot:spring-boot-starter-web')
 	compile('com.google.apis:google-api-services-calendar:v3-rev271-1.23.0')
 	runtime('org.springframework.boot:spring-boot-devtools')
 	runtime('mysql:mysql-connector-java')
 	compileOnly('org.projectlombok:lombok')
+	compile group: 'com.squareup.retrofit2', name: 'retrofit', version: '2.3.0'
```
Comment (adamgyulavari) [inline build.gradle]: styling, all dependencies should look the same

File: src/main/java/com/greenfoxacademy/opal/kalendaryo/kalendaryo/KalendaryoApplication.java [hunk 15]
```diff
@@ -1,14 +1,21 @@
...
 
 	public static void main(String[] args) {
 		SpringApplication.run(KalendaryoApplication.class, args);
+	}
+
+	@Override
+	public void run(String... args) throws Exception {
 
+//		Flyway flyway = new Flyway();
+//		flyway.setDataSource("jdbc:mysql://localhost:3306/kalendaryo_backend", "root", "12345");
```
Comment (adamgyulavari) [inline src/main/java/com/greenfoxacademy/opal/kalendaryo/kalendaryo/KalendaryoApplication.java]: nobody gets passed with outcommented code on a code review

File: src/main/java/com/greenfoxacademy/opal/kalendaryo/kalendaryo/controllers/NotificationController.java [hunk 16]
```diff
@@ -26,34 +22,42 @@
...
     public HttpStatus eventNotification(HttpServletRequest request) {
-
         EventResponse eventResponse = new EventResponse(request);
 
-        if (eventResponse.validate().equals(HttpStatus.OK)) {
+        if (!eventResponse.validate()) {
             eventResponseService.saveEventResponse(eventResponse);
+            return HttpStatus.OK;
         } else {
             System.out.println("Missing: " + eventResponse.getMissingFields());
```
Comment (adamgyulavari) [inline src/main/java/com/greenfoxacademy/opal/kalendaryo/kalendaryo/controllers/NotificationController.java]: let's start to use `logger`

File: src/main/java/com/greenfoxacademy/opal/kalendaryo/kalendaryo/model/EventResponse.java [hunk 17]
```diff
@@ -15,15 +15,15 @@
 
     @Id
     @GeneratedValue(strategy = GenerationType.AUTO)
-    Long eventResponseId;
+    Long id;
 
-    //The channel ID
+    // The channel ID
```
Comment (adamgyulavari) [inline src/main/java/com/greenfoxacademy/opal/kalendaryo/kalendaryo/model/EventResponse.java]: thx 👍

File: src/main/java/com/greenfoxacademy/opal/kalendaryo/kalendaryo/model/EventResponse.java [hunk 18]
```diff
@@ -15,15 +15,15 @@
...
     String channelId;
 
-    //It identifies the watched resource, in our case the event, ID of the event
+    // It identifies the watched resource, in our case the event, ID of the event
```
Comment (adamgyulavari) [inline src/main/java/com/greenfoxacademy/opal/kalendaryo/kalendaryo/model/EventResponse.java]: pls remove me

File: src/main/java/com/greenfoxacademy/opal/kalendaryo/kalendaryo/model/EventResponse.java [hunk 19]
```diff
@@ -50,11 +50,11 @@ public EventResponse(HttpServletRequest request) {
     }
 
     public Long getEventResponseId() {
```
Comment (adamgyulavari) [inline src/main/java/com/greenfoxacademy/opal/kalendaryo/kalendaryo/model/EventResponse.java]: the `id` field should have a `getId` getter

File: src/main/java/com/greenfoxacademy/opal/kalendaryo/kalendaryo/model/UserModel.java [hunk 20]
```diff
@@ -1,25 +1,36 @@
...
     }
 
-    public UserModel(String clientToken) {
-        this.clientToken = clientToken;
+
+    public String getRandomClientToken() {
+        SecureRandom secureRandom = new SecureRandom();
+        byte[] random = new byte[20];
+        secureRandom.nextBytes(random);
+        return Base64.encodeBase64String(random);
```
Comment (adamgyulavari) [inline src/main/java/com/greenfoxacademy/opal/kalendaryo/kalendaryo/model/UserModel.java]: lets move this to a service

File: src/main/java/com/greenfoxacademy/opal/kalendaryo/kalendaryo/model/UserModel.java [hunk 21]
```diff
@@ -1,25 +1,36 @@
...
 import java.util.List;
 
 @Entity
 public class UserModel {
 
     @Id
-    @GeneratedValue(strategy = GenerationType.AUTO)
     private long id;
-
     private String clientToken;
```
Comment (adamgyulavari) [inline src/main/java/com/greenfoxacademy/opal/kalendaryo/kalendaryo/model/UserModel.java]: this field should be `unique`

File: src/main/java/com/greenfoxacademy/opal/kalendaryo/kalendaryo/repository/UserModelRepository.java [hunk 22]
```diff
@@ -1,10 +1,12 @@
...
-import com.greenfoxacademy.opal.kalendaryo.kalendaryo.model.AuthModel;
+
 import com.greenfoxacademy.opal.kalendaryo.kalendaryo.model.UserModel;
 import org.springframework.data.repository.CrudRepository;
 
 public interface UserModelRepository extends CrudRepository<UserModel, Long> {
 
     UserModel findById(Long userId);
+
+    UserModel findAllByClientToken(String clienttoken);
```
Comment (adamgyulavari) [inline src/main/java/com/greenfoxacademy/opal/kalendaryo/kalendaryo/repository/UserModelRepository.java]: this should be findOne

File: build.gradle (hunk 14 above)
Comment (Phaedra144) [reply build.gradle]: we don't need them actually

File: src/main/java/com/greenfoxacademy/opal/kalendaryo/kalendaryo/model/UserModel.java (hunk 20 above)
Comment (adamgyulavari) [reply src/main/java/com/greenfoxacademy/opal/kalendaryo/kalendaryo/model/UserModel.java]: ping

**Pull request merged**